qiskit_experiments/curve_analysis/visualization/fit_result_plotters.py

```diff
@@ -161,16 +148,175 @@ def draw(
                 scaled_maxv, prefix = detach_prefix(maxv, decimal=3)
                 prefactor = scaled_maxv / maxv
                 # pylint: disable=cell-var-from-loop
-                sub_axis.set_major_formatter(FuncFormatter(lambda x, p: f"{x * prefactor: g}"))
+                sub_axis.set_major_formatter(FuncFormatter(lambda x, p: f"{x * prefactor: .3g}"))
```

Comment:
Why did you switch from the `f` format specifier to `g`? I thought with the way that you determined the prefix and scaled the data above that `f` would work here.

Reply:
I noticed `.f` sometime becomes very ugly if the number has three digit, e.g. `700.000`

Reply:
Okay, that makes sense. With the way you scaled the numbers I think `0.3g` should be good. The maximum tick should be less than 999. The maximum tick could be close to 1 and then the smaller ticks would be less than 1 so you need the decimal that g can give when necessary.